extensions/amp-carousel/0.1/slidescroll.js

```diff
@@ -171,10 +191,18 @@ export class AmpSlideScroll extends BaseCarousel {
    * @private
    */
   hideRestOfTheSlides_(index) {
-    for (let i = 0; i < this.noOfSlides_; i++) {
+    const noOfSlides_ = this.noOfSlides_;
+    for (let i = 0; i < noOfSlides_; i++) {
       if (i != index && i != index - 1 && i != index + 1 &&
           this.slideWrappers_[i]) {
         if (this.slideWrappers_[i].classList.contains(SHOWN_CSS_CLASS)) {
+          if (this.hasLooping_) {
+            if ((index == 0 && i == noOfSlides_ - 1) ||
+                (index == noOfSlides_ - 1 && i == 0)) {
+              continue;
```

Comment:
What does this `continue` means? That this slide should be visible? If so - this is duplication of logic from `showSlide_`. We should instead save points to prev and next slide and test them here. In either case, the `continue` so deep in `if`s is dangerous.